File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-3.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    void traversal(TreeNode*root, TreeNode*p,TreeNode*q,TreeNode* &ans)
    {
        
        if(p->val == root->val || q->val==root->val)
        {
            if(p->val == root->val)
            {
                ans = p;
            }
            else
            {
                ans = q;
            }
        }
        if((p->val < root->val && q->val > root->val) || (q->val < root->val && p->val > root->val))
        {
            ans = root;
        }
        if(p->val > root->val && q->val > root->val)
        {
            traversal(root->right,p,q,ans);
        }
        if(p->val < root->val && q->val < root->val)
        {
            traversal(root->left,p,q,ans);
        } 

        return;
    }
    TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
        TreeNode* ans = new TreeNode(101);
        traversal(root,p,q,ans);
        return ans;
        

        
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        self.ans = True
        def depth(root):
            if self.ans == False:
                return 0
            if root is None:
                return 0
            
            ld = depth(root.left)
            rd = depth(root.right)
            print(ld-rd)
            if abs(ld-rd) > 1:
                self.ans = False
            return max(ld,rd) + 1

        depth(root)
        return self.ans
        